groups/bsl/bsls/bsls_atomicoperations_all_all_clangintrinsics.h

```diff
@@ -0,0 +1,454 @@
+// bsls_atomicoperations_all_all_clangintrinsics.h                    -*-C++-*-
+#ifndef INCLUDED_BSLS_ATOMICOPERATIONS_ALL_ALL_CLANGINTRINSICS
+#define INCLUDED_BSLS_ATOMICOPERATIONS_ALL_ALL_CLANGINTRINSICS
+
+#ifndef INCLUDED_BSLS_IDENT
+#include <bsls_ident.h>
+#endif
+BSLS_IDENT("$Id: $")
+
+//@PURPOSE: Implement atomic operations using clang '__c11_atomic_*' intrinsics
+//
+//@CLASSES:
+//  bsls::AtomicOperations_ALL_ALL_ClangIntrinsics:
+//  clang '__c11_atomic_*' instrinsics
+//
+//@DESCRIPTION: This component provides classes necessary to implement atomics
+// in 32bit/64bit mode using clang compiler intrinsics.  The classes are for
+// private use only.  See 'bsls_atomicoperations' and 'bsls_atomic' for the
+// public interface to atomics.
+
+#ifndef INCLUDED_BSLS_ATOMICOPERATIONS_DEFAULT
+#include <bsls_atomicoperations_default.h>
+#endif
+
+#ifndef INCLUDED_BSLS_PLATFORM
+#include <bsls_platform.h>
+#endif
+
+#ifndef INCLUDED_BSLS_TYPES
+#include <bsls_types.h>
+#endif
+
+#if defined(BSLS_PLATFORM_CMP_CLANG)
+#if __has_extension(c_atomic) || __has_extension(cxx_atomic)
+
+namespace BloombergLP {
+
+namespace bsls {
+
+struct AtomicOperations_ALL_ALL_ClangIntrinsics;
+typedef AtomicOperations_ALL_ALL_ClangIntrinsics  AtomicOperations_Imp;
+
+     // ==================================================================
+     // struct Atomic_TypeTraits<AtomicOperations_ALL_ALL_ClangIntrinsics>
+     // ==================================================================
+
+template <>
+struct Atomic_TypeTraits<AtomicOperations_ALL_ALL_ClangIntrinsics>
+{
+    struct __attribute__((__aligned__(sizeof(int)))) Int
+    {
+          _Atomic(int) d_value;
```

Comment:
I think these are expressed in terms of clang implementation, and the intended public typdef is:
`atomic_int`